src/fp_liste.cpp

```diff
@@ -20,25 +20,23 @@
 #include "fp_liste.h"
 //#include<iostream>
 //using namespace std;
-void Liste::getMin(double *min_, int *origine_, double *mean_){
+void Liste::getMin(double *min_, int *origine_, double *mean_, int *intervals_){
 	
 	Liste* Current = this;	
 	min_[0] = Current->cFunc->min();
 	origine_[0] = Current->cFunc->origine;
 	mean_[0] = Current->cFunc->mean();
-	//cout << "St. Origine : " << origine_[0] << "Cost min : " << min_[0] << 
-	//"[" << Current->cFunc->lBorder <<  "," << Current->cFunc->rBorder << "]" << "\n";
+	intervals_[0] = 1;
 	double tmp=0.;
 	while(Current-> next != NULL){
 		Current = Current->next;
 		tmp = Current->cFunc->min();
-		//cout << "It. Origine : " << Current->cFunc->origine << "Cost min : " << tmp << 
-		//"[" << Current->cFunc->lBorder << "," << Current->cFunc->rBorder << "]" << "\n";
 		if(min_[0] > tmp){
 			min_[0] = tmp;
 			origine_[0] = Current->cFunc->origine;	
 			mean_[0] = Current->cFunc->mean();	
 		}
+		intervals_[0] += 1;
 	}
 }
```

Comment:
I think it would be easier to have a dedicated function to count the number of intervals.
Something like:
void Liste::getNbIntervals(int *intervals_){
	Liste* Current = this;	
	while(Current-> next != NULL){
		intervals_[0] += 1;
	}
}

Reply:
my intent was to make a small modification of the existing code, without adding any new functions/methods, but that is fine to add a new method, if you prefer.